Lib/test/test_httpservers.py

```diff
@@ -539,6 +539,51 @@ def test_get(self):
             finally:
                 os.chmod(self.tempdir, 0o755)
 
+    def test_range_get(self):
+        response = self.request(self.base_url + '/test')
```

Comment:
All the URLs are `base_url/test`, so I'd suggest creating a variable for that, say `route = self.base_url + '/test'` or `route = f'{self.base_url}/test`. In addition, can you check what happens when you specify an incorrect method (POST instead of GET, etc).

Reply:
`HEAD` should always return the same headers with `GET` but without actual content, I didn't touch `HEAD` related code so I didn't add tests for that. It should work just well.